pkg/batch/executor_test.go

```diff
@@ -115,12 +117,15 @@ func testReadAfterWrite(t *testing.T) {
 	// following that write, another reader starts, and must read the updated value
 	go func() {
 		<-write1Done // ensure we start AFTER write1 has completed
-		r := reader{Ch: read2Batched, Fn: func() (interface{}, error) {
-			return nil, nil
-		}}
-		_, _ = exec.BatchFor("k", 50*time.Millisecond, &r)
-		// We expect r2's exec function to no execute because it should join r1's batch
-		if r.IsExecuted || db.GetAccessCount() != accessedOnce {
+		te := trackableExecuter{batchTracker: read2Batched, execTracker: make(chan struct{})}
+		r2, _ := exec.BatchFor("k", 50*time.Millisecond, &te)
+		r2v := r2.(string)
+		if r2v != "v1" {
+			t.Errorf("expected r2 to get v1, got %s instead", r2v)
+		}
+
+		// We expect r2's exec function to not execute because it should join r1's batch
+		if te.WasExecuted() || db.GetAccessCount() != 1 {
```

Comment:
can you split the check into 2 if(s) so when it will fail we will know which one failed or capture the result of the check and print it